app/javascript/history/history.jsx

```diff
@@ -1,213 +1,106 @@
 import { h, Component } from 'preact';
 import { PropTypes } from 'preact-compat';
 import debounce from 'lodash.debounce';
-import setupAlgoliaIndex from '../src/utils/algolia';
+
+import {
+  defaultState,
+  loadNextPage,
+  onSearchBoxType,
+  performInitialSearch,
+  search,
+  toggleTag,
+} from '../searchableItemList/searchableItemList';
+import { ItemListLoadMoreButton } from '../src/components/ItemList/ItemListLoadMoreButton';
+import { ItemListTags } from '../src/components/ItemList/ItemListTags';
+import { ItemListItem } from '../src/components/ItemList/ItemListItem';
```

Comment:
Again, in the spirit of composition I've extracted a few components shared by both pages